src/routes/conversation/[id]/+server.ts

```diff
@@ -250,13 +250,31 @@ export async function POST({ request, locals, params, getClientAddress }) {
 
 			let webSearchResults: WebSearch | undefined;
 
-			if (webSearch && !isContinue && !conv.assistantId) {
-				webSearchResults = await runWebSearch(conv, messages[messages.length - 1].content, update);
+			// check if assistant has a rag
+			const assistantRAG = await collections.assistants
+				.findOne({ _id: conv.assistantId })
```

Comment:
if we are only using `rag` field, we can use [projection](https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/) ?

Is using projection important here? (cc: @coyotte508)

Reply:
projection is always a nice small perf boost

Just make sure to always change the return typing if you use projection:

```suggestion
			const assistantRAG = await collections.assistants
				.findOne<Pick<..., "rag">>({ _id: conv.assistantId, {projection: {rag: 1} })
```

Reply:
fixed!